### PR TITLE
[Fix #847] Allow ASRSnippets to get marked `for_qa`.

### LINKED ISSUE #847: Allow ASRSnippets to get marked `for_qa`
For #845 we want to have a set of snippets on both Production and Staging that will be used for QA. 

To avoid accidental changes to those snippets and also to avoid confusing Snippet Editos, those ASRSnippets will be only visible to superadmins.

A new ASRSnippet field `for_qa`, editable only by superadmins, will identify snippets for QA.

### PR DESCRIPTION
**Pull request by glogiotatidis**

### CODE REVIEW

**Comment by glogiotatidis** [comment]
Hey @jpetto this is a blocker for #845. Let me know if you don't have the bw to review this today. Many thanks!

**Comment by glogiotatidis** [comment]
hey @akatsoulas maybe you can take a look since this need to be out today?

File: snippets/base/admin/adminmodels.py [hunk 1]
```diff
@@ -223,6 +223,18 @@ def change_view(self, request, *args, **kwargs):
...
+        if not request.user.is_superuser:
+            return self.readonly_fields + ('for_qa',)
+        return self.readonly_fields
+
+    def get_queryset(self, request):
+        queryset = super().get_queryset(request)
+        if request.user.is_superuser:
+            return queryset
+        else:
+            return queryset.exclude(for_qa=True)
```
Comment (akatsoulas) [inline snippets/base/admin/adminmodels.py]: nit: I would go for `filter(for_qa=False)` since it might be easier to read instead of using a `NOT` of sorts.

**Comment by glogiotatidis** [comment]
Thanks!

**Pull request merged**